test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go

```diff
@@ -0,0 +1,113 @@
+package pao_mustgather
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+
+	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/nodes"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profiles"
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var mustGatherPath = os.Getenv("MUSTGATHER_DIR")
+var mustGatherFailed bool
+var mustGatherContentDir string
+
+var _ = Describe("Testing PAO Mustgather", func() {
+	var profile *performancev2.PerformanceProfile
+	var workerRTNodes []corev1.Node
+	var err error
+	if mustGatherPath == "" {
+		mustGatherFailed = true
+	}
+
+	testutils.BeforeAll(func() {
+		if mustGatherFailed {
```

Comment:
I'd set the `mustGatherPath` here and skip directly, avoiding `mustGatherFailed`.
```
mustGatherPath = os.Getenv("MUSTGATHER_DIR")
var mustGatherFailed bool
if mustGatherPath == "" {
  Skip("No mustgather directory provided, check MUSTGATHER_DIR environment variable")
}
// avoid else
```

Reply:
done